src/main/java/frc/robot/Systems/Chassis/Chassis.java

```diff
@@ -124,34 +166,40 @@ public void setGyroRot(double rot) {
   }
 
   public SwerveModulePosition[] getPositions () {
-    SwerveModulePosition[] positions = new SwerveModulePosition[modules.length];
     for (int i = 0; i < modules.length; i++) {
-      positions[i] = modules[i].getPosition();
+      getPositions[i] = modules[i].getPosition();
+    }
```

Comment:
I wonder if it would be better to update the 'actual positions' array in `periodic`, and this is just a normal getter that returns that array.  🤔